perceval/backends/core/pagure.py

```diff
@@ -0,0 +1,418 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Animesh Kumar <animuz111@gmail.com>
+#
+
+import json
+import logging
+import requests
+from grimoirelab_toolkit.datetime import (str_to_datetime)
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser,
+                        DEFAULT_SEARCH_FIELD)
+from ...client import HttpClient
+from ...utils import DEFAULT_DATETIME, DEFAULT_LAST_DATETIME
+from datetime import datetime
+
+CATEGORY_ISSUE = "issue"
+
+PAGURE_URL = "https://pagure.io/"
+PAGURE_API_URL = "https://pagure.io/api/0"
+
+MAX_CATEGORY_ITEMS_PER_PAGE = 100
+PER_PAGE = 100
+
+# Default sleep time and retries to deal with connection/server problems
+DEFAULT_SLEEP_TIME = 1
+MAX_RETRIES = 5
+
+logger = logging.getLogger(__name__)
+
+
+class Pagure(Backend):
+    """Pagure backend for Perceval.
+
+    This class allows the fetch the issues stored in a Pagure
+    repository. Note that api token is needed to perform
```

Comment:
Is the sentence `Note that ...` true? Which API calls can be done without/with authentication?

Reply:
In the current implementation, none of the APIs really require a token. However, since the APIs did accept a token as a header I had included it. I have removed this statement from the doc.